program/tests/test_stuck_escrow.rs

```diff
@@ -0,0 +1,491 @@
+}
+
+#[test]
+fn test_close_stuck_escrow_fails_wrapped_mint_authority_mismatch() {
+    let incorrect_authority = Pubkey::new_unique();
+
+    CloseStuckEscrowBuilder::default()
+        .wrapped_mint_authority(incorrect_authority)
+        .check(Check::err(TokenWrapError::MintAuthorityMismatch.into()))
+        .execute();
+}
+
+#[test]
+fn test_close_stuck_escrow_fails_escrow_mismatch() {
+    let incorrect_escrow = KeyedAccount {
+        key: Pubkey::new_unique(), // Not the derived ATA
+        account: Account {
+            owner: spl_token_2022::id(),
+            lamports: Rent::default().minimum_balance(spl_token_2022::state::Account::LEN),
+            ..Default::default()
+        },
+    };
+    CloseStuckEscrowBuilder::default()
+        .escrow_account(incorrect_escrow)
+        .check(Check::err(TokenWrapError::EscrowMismatch.into()))
+        .execute();
+}
+
+#[test]
+fn test_close_stuck_escrow_fails_escrow_owner_mismatch() {
+    let unwrapped_token_program = TokenProgram::SplToken2022;
+    let wrapped_token_program = TokenProgram::SplToken2022;
+
+    let unwrapped_mint = KeyedAccount {
+        key: Pubkey::new_unique(),
+        account: setup_mint(
+            unwrapped_token_program,
+            &Rent::default(),
+            Pubkey::new_unique(),
+        ),
+    };
+
+    let wrapped_mint = KeyedAccount {
+        key: get_wrapped_mint_address(&unwrapped_mint.key, &wrapped_token_program.id()),
+        account: setup_mint(
+            wrapped_token_program,
+            &Rent::default(),
+            Pubkey::new_unique(),
+        ),
+    };
+
+    // This is the correct PDA that *should* own the escrow account's tokens.
+    let wrapped_mint_authority = get_wrapped_mint_authority(&wrapped_mint.key);
+    let expected_escrow_address = get_escrow_address(
+        &unwrapped_mint.key,
+        &unwrapped_token_program.id(),
+        &wrapped_token_program.id(),
+    );
+
+    // Create an escrow account at the correct ATA, but set its internal owner field
+    // to a random, incorrect public key.
+    let account_size = spl_token_2022::state::Account::LEN;
+    let mut account_data = vec![0; account_size];
+    let mut state =
+        PodStateWithExtensionsMut::<PodAccount>::unpack_uninitialized(&mut account_data).unwrap();
+    state.base.mint = unwrapped_mint.key;
+    state.base.owner = Pubkey::new_unique();
+    state.base.amount = 0.into();
+    state.base.state = AccountState::Initialized.into();
+    state.init_account_type().unwrap();
+
+    let escrow_with_wrong_owner = KeyedAccount {
+        key: expected_escrow_address,
+        account: Account {
+            lamports: Rent::default().minimum_balance(account_size),
+            data: account_data,
+            owner: spl_token_2022::id(),
+            ..Default::default()
+        },
+    };
+
+    CloseStuckEscrowBuilder::default()
+        .unwrapped_token_program(unwrapped_token_program)
+        .unwrapped_token_program(wrapped_token_program)
```

Comment:
right?
```suggestion
        .wrapped_token_program(wrapped_token_program)
```

Reply:
What in the world (?), got lucky on the test passing given the default happens to be the correct value. Will fix.